File: test-suite/src/test/java/io/micronaut/views/model/security/MockAuthenticationProvider.java
package io.micronaut.views.model.security;

import io.micronaut.core.annotation.Nullable;
import io.micronaut.http.HttpRequest;
import io.micronaut.security.authentication.AuthenticationRequest;
import io.micronaut.security.authentication.AuthenticationResponse;
import io.micronaut.security.authentication.provider.HttpRequestAuthenticationProvider;
import jakarta.inject.Singleton;

import java.util.Collections;

@Singleton
public class MockAuthenticationProvider<B> implements HttpRequestAuthenticationProvider<B> {

    @Override
    public AuthenticationResponse authenticate(@Nullable HttpRequest<B> httpRequest, AuthenticationRequest<String, String> authenticationRequest) {
        return AuthenticationResponse.success(authenticationRequest.getIdentity(), Collections.singletonMap("email", "john@email.com"));
    }
}

File: test-suite-python/src/test/python/micronaut/views/model/security/MockAuthenticationProvider.py
from jakarta.inject import Singleton
from micronaut.http import HttpRequest
from micronaut.security.authentication import AuthenticationRequest, AuthenticationResponse
from micronaut.security.authentication.provider import HttpRequestAuthenticationProvider


@Singleton
class MockAuthenticationProvider(HttpRequestAuthenticationProvider):

    def authenticate(self, httpRequest: HttpRequest | None, authenticationRequest: AuthenticationRequest) -> AuthenticationResponse:
        return AuthenticationResponse.success(authenticationRequest.getIdentity(), {"email": "john@email.com"})
